File: NewsCollector/src/com/twovix/newscollector/RssPlugin.java
package com.twovix.newscollector;


public abstract class RssPlugin {

	private String path = "";

	public String getPath() {
		return path;
	}
	
	public RssPlugin(String rpath){
		path = rpath;
	}
	
	protected String getLink(String link) {
		return link;
	}
}

File: NewsCollector/src/com/twovix/newscollector/RssPlugin.scala
package com.twovix.newscollector

abstract class RssPlugin (path:String) {

	def getPath():String =  {
		return path;
	}

	def getLink(link:String):String 
}

class RssPluginGoogle (path:String) extends RssPlugin (path) {
  	override def getLink(link:String):String = {	
  		link.split("url=")(1);
  	}
}

class RssPluginStandard  (path:String) extends RssPlugin (path) {
  
  override def getLink(link:String):String = {
	 link
  }
}